src/pages/index.js

```diff
@@ -124,4 +125,93 @@ const enableValidation = (config) => {
   });
 };
 
+//initializations
+
+const user = new UserInfo(userData);
+
+const profilePopup = new PopupWithForm(editProfilePopup, {
+  handleFormSubmit: ({ name, about_me }) => {
+    const profileInfoSubmit = document.querySelector("#profile-info-submit");
+    profileInfoSubmit.textContent = "Saving...";
+    user.setUserInfo({ newName: name, newJob: about_me });
+    api.patchInfo({name, about: about_me});
+    setTimeout(profilePopup.close(), 300); //I am aware this isn't necessarily the same thing as having it close when the api is finished- this is a simulacrum of it. I couldn't figure out how to connect it all. You will see this replicated elsewhere
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well